### PR TITLE
check consumer auth disabled field

### LINKED ISSUE #309: disabled:true doesn't disable ConsumerAuthorizations at the operation level

### PR DESCRIPTION
**Pull request by rockspore**
close #309

### CODE REVIEW

**Comment by theganyo** [comment]
ConsumerAuthorization.isEmpty() already checks the disabled flag. https://github.com/apigee/apigee-remote-service-envoy/blob/master/config/env_spec_ext.go#L136

**Comment by rockspore** [comment]
> ConsumerAuthorization.isEmpty() already checks the disabled flag. https://github.com/apigee/apigee-remote-service-envoy/blob/master/config/env_spec_ext.go#L136

Ah I missed it. Then it must be a different bug somewhere. I will take a look

**Comment by theganyo** [comment]
What is the actual bug? The issue is prescriptive, not descriptive.

File: config/env_spec_request.go [hunk 1]
```diff
@@ -321,7 +321,7 @@ func (e *EnvironmentSpecRequest) getAuthenticationRequirement() (auth Authentica
 
 // IsAuthorizationRequired returns true if Authorization is required.
 func (e *EnvironmentSpecRequest) IsAuthorizationRequired() bool {
-	return !e.GetConsumerAuthorization().isEmpty()
+	return !e.GetConsumerAuthorization().isEmpty() && !e.GetConsumerAuthorization().Disabled
```
Comment (theganyo) [inline config/env_spec_request.go:324]: Reversing the checks would be more efficient.

Comment (rockspore) [reply config/env_spec_request.go]: Done. Thanks.

**Review by theganyo — APPROVED** [review]

**Pull request merged**